src/gradata/enhancements/behavioral_extractor.py

```
@@ -0,0 +1,497 @@
+
+    draft_sents = _split_sentences(draft)
+    final_sents = _split_sentences(final)
+    draft_words = set(draft.lower().split())
+    final_words = set(final.lower().split())
+    added_words = final_words - draft_words
+    removed_words = draft_words - final_words
+
+    # Precompute word sets for sentence overlap (avoids re-splitting per pair)
+    draft_sent_sets = [set(s.lower().split()) for s in draft_sents]
+    final_sent_sets = [set(s.lower().split()) for s in final_sents]
+    added_sents = [s for s, ws in zip(final_sents, final_sent_sets)
```

Ruff (0.15.9)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)